script.module.clouddrive.common/clouddrive/common/ui/addon.py

```diff
@@ -548,14 +614,15 @@ def play(self, driveid, item_driveid=None, item_id=None):
             for subtitle in item['subtitles']:
                 subtitles.append(DownloadServiceUtil.build_download_url(driveid, Utils.default(Utils.get_safe_value(subtitle, 'drive_id'), driveid), subtitle['id'], urllib.quote(Utils.str(subtitle['name']))))
             list_item.setSubtitles(subtitles)
-        xbmcplugin.setResolvedUrl(self._addon_handle, True, list_item)
+        if not self.cancel_operation():
+            xbmcplugin.setResolvedUrl(self._addon_handle, True, list_item)
```

Comment:
same, dependency on xbmcplugin does not make sense on a script.module

Reply:
Same explanation, the addon will inherit this class.